### PR TITLE
Fixes #511

### LINKED ISSUE #511: Loading a mergedGrid does not work in current MoBIE-beta
Loading a dataset that contains a mergedGrid fails with:
```
java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.RuntimeException: Module threw exception
	at net.imagej.legacy.LegacyService.runLegacyCompatibleCommand(LegacyService.java:307)
	at net.imagej.legacy.DefaultLegacyHooks.interceptRunPlugIn(DefaultLegacyHooks.java:166)
	at ij.IJ.runPlugIn(IJ.java)
	at ij.Executer.runCommand(Executer.java:150)
	at ij.Executer.run(Executer.java:68)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.util.concurrent.ExecutionException: java.lang.RuntimeException: Module threw exception
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at net.imagej.legacy.LegacyService.runLegacyCompatibleCommand(LegacyService.java:303)
	... 5 more
Caused by: java.lang.RuntimeException: Module threw exception
	at org.scijava.module.ModuleRunner.call(ModuleRunner.java:127)
	at org.scijava.module.ModuleRunner.call(ModuleRunner.java:63)
	at org.scijava.thread.DefaultThreadService.lambda$wrap$2(DefaultThreadService.java:225)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	... 1 more
Caused by: java.lang.RuntimeException: Unknown class: mergedGrid
	at org.embl.mobie.viewer.serialize.JsonHelper.createObjectFromJsonValue(JsonHelper.java:45)
	at org.embl.mobie.viewer.serialize.SourceTransformerListAdapter.deserialize(SourceTransformerListAdapter.java:38)
	at org.embl.mobie.viewer.serialize.SourceTransformerListAdapter.deserialize(SourceTransformerListAdapter.java:14)
	at com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222)
	at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:41)
	at com.google.gson.internal.bind.MapTypeAdapterFactory$Adapter.read(MapTypeAdapterFactory.java:187)
	at com.google.gson.internal.bind.MapTypeAdapterFactory$Adapter.read(MapTypeAdapterFactory.java:145)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222)
	at com.google.gson.Gson.fromJson(Gson.java:932)
	at com.google.gson.Gson.fromJson(Gson.java:897)
	at com.google.gson.Gson.fromJson(Gson.java:846)
	at org.embl.mobie.viewer.serialize.DatasetJsonParser.parseDataset(DatasetJsonParser.java:24)
	at org.embl.mobie.viewer.MoBIE.openDataset(MoBIE.java:220)
	at org.embl.mobie.viewer.MoBIE.openDataset(MoBIE.java:160)
	at org.embl.mobie.viewer.MoBIE.<init>(MoBIE.java:96)
	at org.embl.mobie.viewer.command.OpenMoBIEProjectCommand.run(OpenMoBIEProjectCommand.java:25)
	at org.scijava.command.CommandModule.run(CommandModule.java:196)
	at org.scijava.module.ModuleRunner.run(ModuleRunner.java:163)
	at org.scijava.module.ModuleRunner.call(ModuleRunner.java:124)
	... 6 more
```

To reproduce: open the project at /g/kreshuk/pape/Work/mobie/covid-if-project (data is not uploaded to s3 yet)

### PR DESCRIPTION
**Pull request by tischi**
@K-Meech 

Could we merge this?

I should fix https://github.com/mobie/mobie-viewer-fiji/issues/511

That is, it should open this without an error:

```
final ImageJ imageJ = new ImageJ();
		imageJ.ui().showUI();

		try {
			new MoBIE("/g/kreshuk/pape/Work/mobie/covid-if-project/data",
					MoBIESettings.settings().imageDataFormat( ImageDataFormat.OmeZarr ) );
		} catch (IOException e) {
			e.printStackTrace();
		}
```

### CODE REVIEW

File: src/main/java/org/embl/mobie/viewer/transform/MergedGridSourceTransformer.java [hunk 1]
```diff
@@ -41,6 +40,11 @@ public void transform( Map< String, SourceAndConverter< ? > > sourceNameToSource
 
 		sourceNameToSourceAndConverter.put( mergedSourceAndConverter.getSpimSource().getName(), mergedSourceAndConverter );
 
+		if ( mergedGridSourceName.equals( "plate_cell_segmentation" ) )
```
Comment (KateMoreva) [inline src/main/java/org/embl/mobie/viewer/transform/MergedGridSourceTransformer.java:43]: We will not need this for other sources as well?

File: src/main/java/org/embl/mobie/viewer/transform/MergedGridSourceTransformer.java [hunk 2]
```diff
@@ -41,6 +40,11 @@ public void transform( Map< String, SourceAndConverter< ? > > sourceNameToSource
...
+		{
+			int a = 1;
```
Comment (K-Meech) [inline src/main/java/org/embl/mobie/viewer/transform/MergedGridSourceTransformer.java:45]: I guess this hardcoding needs to be removed for 'plate_cell_segmentation'?

**Review by K-Meech — APPROVED** [review]
Looks good! I just noticed you have one hardcoded part for 'plate_cell_segmentation'. I guess that needs to be removed?

**Review by KateMoreva — APPROVED** [review]

File: src/main/java/org/embl/mobie/viewer/transform/MergedGridSourceTransformer.java (hunk 2 above)
Comment (tischi) [reply src/main/java/org/embl/mobie/viewer/transform/MergedGridSourceTransformer.java]: Good catch! I removed it.

**Pull request merged**